src/main/kotlin/com/hypto/iam/server/service/UsersService.kt

```diff
@@ -350,16 +350,19 @@ class UsersServiceImpl : KoinComponent, UsersService {
             user.createdBy,
             true,
         )
-        userAuthRepo.create(
-            hrn = user.hrn,
-            providerName = TokenServiceImpl.ISSUER,
-            authMetadata = null,
-        )
-        userRepo.update(
-            hrn = user.hrn,
-            verified = true,
-            loginAccess = true,
-        )
+        txMan.wrap {
+            userAuthRepo.create(
+                hrn = user.hrn,
+                providerName = TokenServiceImpl.ISSUER,
+                authMetadata = null,
+            )
+            userRepo.update(
+                hrn = user.hrn,
+                verified = true,
+                loginAccess = true,
+            )
+            passcodeRepo.deleteByEmailAndPurpose(user.email!!, VerifyEmailRequest.Purpose.reset)
```

Comment:
is purpose always reset ?

Reply:
I have changed it to `invite`, and this function is called at one place where we check a passcode for `invite` purpose and then call this function